### DISCUSSION TITLE
Issue with pull from upstream

### DISCUSSION THREAD

**Discussion by javadzarezadeh** [fastapi/full-stack-fastapi-template#1290] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
git pull --no-commit upstream master
```

### Description

After I try `git pull --no-commit upstream master` the terminal output would be a fatal error of:

<img width="566" alt="image" src="https://github.com/user-attachments/assets/94c0e3d4-e3d3-4666-8b91-1253093d39a6">

After Googling my issue I found [this](https://stackoverflow.com/questions/62653114/how-can-i-deal-with-this-git-warning-pulling-without-specifying-how-to-reconci), but since I don't have any experience with updating from an upstream, I don't know which procedure should I choose. Would you please help me through this?

Thank you.

### Operating System

macOS

### Operating System Details

Sonoma 14.5

### Python Version

3.12.2

**Comment by YuriiMotov** [comment, marked as answer]
Use
```
git fetch upstream master
git merge --no-commit --no-ff upstream/master
```
This will merge changes, but stop before commit so that you can check everything
